src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs

```
@@ -99,4 +99,12 @@ sorting switch
                 companyCertificate.ValidTill
                 ))
         .SingleOrDefaultAsync();
+
+    public Task<(byte[] Content, string FileName, MediaTypeId MediaTypeId, bool IsExist)> GetCompanyCertificateDocumentDataAsync(Guid documentId, DocumentTypeId documentTypeId) =>
+            _context.Documents
+            .Where(x => x.Id == documentId &&
+                   x.DocumentStatusId == DocumentStatusId.LOCKED &&
```

You are selecting only locked documents, it'd be better to don't filter for the status here to or select a bool value which provides the information if the document is in status locked. if it isn't than please return a specific error message

---

filter for status removed and a bool check is added